src/optimagic/optimization/multistart_options.py

```diff
@@ -0,0 +1,482 @@
+        )
+
+    if (
+        not isinstance(options.mixing_weight_bounds, tuple)
+        or len(options.mixing_weight_bounds) != 2
+        or not set(type(x) for x in options.mixing_weight_bounds) <= {int, float}
+    ):
+        raise InvalidMultistartError(
+            f"Invalid mixing weight bounds: {options.mixing_weight_bounds}. Mixing "
+            "weight bounds must be a tuple of two numbers."
+        )
+
+    if options.convergence_xtol_rel is not None and (
+        not isinstance(options.convergence_xtol_rel, int | float)
+        or options.convergence_xtol_rel < 0
+    ):
+        raise InvalidMultistartError(
+            "Invalid relative params tolerance:"
+            f"{options.convergence_xtol_rel}. Relative params "
+            "tolerance must be a number."
+        )
+
+    if (
+        not isinstance(options.convergence_max_discoveries, int | float)
+        or options.convergence_max_discoveries < 1
+    ):
+        raise InvalidMultistartError(
+            f"Invalid max discoveries: {options.convergence_max_discoveries}. Max "
+            "discoveries must be a positive integer or infinity."
+        )
+
+    if not isinstance(options.n_cores, int) or options.n_cores < 1:
+        raise InvalidMultistartError(
+            f"Invalid number of cores: {options.n_cores}. Number of cores "
+            "must be a positive integer."
+        )
+
+    if not callable(options.batch_evaluator) and options.batch_evaluator not in (
+        "joblib",
+        "pathos",
+    ):
+        raise InvalidMultistartError(
+            f"Invalid batch evaluator: {options.batch_evaluator}. Batch evaluator "
+            "must be a Callable or one of 'joblib' or 'pathos'."
+        )
+
+    if options.batch_size is not None and (
+        not isinstance(options.batch_size, int) or options.batch_size < options.n_cores
+    ):
+        raise InvalidMultistartError(
+            f"Invalid batch size: {options.batch_size}. Batch size "
+            "must be a positive integer larger than n_cores, or None."
+        )
+
+    if not isinstance(options.seed, int | np.random.Generator | None):
+        raise InvalidMultistartError(
+            f"Invalid seed: {options.seed}. Seed "
+            "must be an integer, a numpy random generator, or None."
+        )
+
+    if options.error_handling is not None and options.error_handling not in (
+        "raise",
+        "continue",
+    ):
+        raise InvalidMultistartError(
+            f"Invalid error handling: {options.error_handling}. Error handling must be "
+            "'raise' or 'continue'."
+        )
+
+
+def _replace_and_warn_about_deprecated_attributes(
+    options: MultistartOptions,
+) -> MultistartOptions:
+    replacements: MultistartOptionsDict = {}
+
+    if options.share_optimization is not None:
+        msg = (
+            "The share_optimization attribute is deprecated and will be removed in "
```

Comment:
For users, "share_optimization" is currently not an attribute but an entry in a dictionary. In general I prefer the word "option" instead of "attribute" for this kind of stuff.